lib/dataset/dataset.cpp

```diff
@@ -277,10 +277,16 @@ typename Masks::holder_type union_or(const Masks &currentMasks,
   for (const auto &[key, item] : otherMasks) {
     if (!currentMasks.contains(key))
       out.insert_or_assign(key, copy(item));
-    else if (out[key].dims().includes(item.dims()))
+    else if (item.dtype() != core::dtype<bool>) {
+      std::string optional_message = " This operation is not supported for "
+                                     "non-boolean masks with same names.";
+      except::throw_mismatch_error(core::dtype<bool>, item.dtype(),
+                                   optional_message);
```

Comment:
I think this is not catching the case where `otherMasks` is `bool` but `currentMasks` is not?

Reply:
You're right, it also doesn't catch if the non-boolean mask in the `otherMasks` is not in the `currentMasks`
I thought it only matters when you want to combine 2 non-boolean masks.

For example, if `data_array_1` has a non-boolean mask called `mask0` and `data_array_2` has a non-boolean mask called `mask1`, then they will be in the result of the operation will just have two non-boolean masks, `mask0` and `mask1`. 

Should we just forbid any non-boolean masks for the binary operations...?

Actually, after you mentioned that, I think it'll be weird to have non-boolean masks when you do `==` operation... Maybe we should just forbid it...?